### PR TITLE
Overview and Getting Started Page for GH Pages.

### PR DESCRIPTION
**Pull request by crdelsey**
## Basic Info

| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | NA |
| Primary OS tested on | Ubuntu |
| Robotic platform tested on |  |

---

## Description of contribution in a few bullet points

* Adds an overview section to the main page and a getting started page that takes you from 0 to navigating in the simplest way possible.

**NOTE** Since the site isn't really considered live yet, I went ahead and published these changes so it is easier to see what the result looks like.

Take a look at this page https://ros-planning.github.io/navigation2/
and the Getting Started page below.

### CODE REVIEW

**Comment by crdelsey** [comment]
I just figured out that, by storing the doc output as artifacts, we can go to the docs_build CircleCI page, artifacts, and open html/index.html to see the site as it appears with the PR changes.

No more need to publish this to the live site just to see what it looks like.

File: sphinx_doc/getting_started/index.rst [hunk 1]
```diff
@@ -3,5 +3,129 @@
...
+      sudo apt install ros2-dashing-navigation2
+      sudo apt install ros2-dashing-nav2-bringup
+
+3. Install the Turtlebot 3 packages. Again, for Ubuntu 18, it looks like this:
+
+   .. code-block:: bash
+
+      sudo apt install ros2-dashing-turtlebot3*
+
+Running the Example
```
Comment (SteveMacenski) [inline sphinx_doc/getting_started/index.rst]: Is there a way to load the markdown file with these same instructions to be viewed? I would ideally not like to have multiple of the same instructions we need to keep in sync.

If we're going the sphinx route, should we just have this be the docs and remove the markdown if we cant link them?

Comment (crdelsey) [reply sphinx_doc/getting_started/index.rst]: > If we're going the sphinx route, should we just have this be the docs and remove the markdown if we cant link them?

Yeah. That was my thinking. Sphinx would be the main set of docs, and the markdown we have would get transitioned to there.

We can put a link in the sphinx docs to point back to github markdown for stuff we haven't transitioned or don't intend to.

Comment (SteveMacenski) [reply sphinx_doc/getting_started/index.rst]: I wasn't sure if you could "render" the markdown file as the webpage (similar to loading a picture or something from the md file). If not, I'd be OK with merging this and then just deleting the md file if they match up.

We  should have the live master sphinx page in the main project readme

**Pull request merged**